core/src/main/scala-3/pureconfig/generic/derivation/EnumConfigReaderDerivation.scala

```diff
@@ -0,0 +1,61 @@
+package pureconfig
+package generic
+package derivation
+
+import scala.compiletime.{constValue, erasedValue, error, summonInline}
+import scala.deriving.Mirror
+
+import pureconfig.error.{CannotConvert, ConfigReaderFailures}
+import pureconfig.generic.derivation.transformedLabelsFor
+import pureconfig.generic.error.NoValidCoproductOptionFound
+
+type EnumConfigReader[A] = EnumConfigReaderDerivation.Default.EnumConfigReader[A]
```

Comment:
The `derived` method needs to be defined in the companion object of the type it derives. Which means if I were to change the return type of the `derived` method here to `ConfigReader[A]`, I would also need to extend `object ConfigReader` with this method – but we already have a different method which does the "normal" derivation for sum types. The user would then have to ensure everything is scoped so they can import just the one `derived` method they need and avoid ambiguous definitions. The other alternative is to embed this derivation into the other extension method.

But because `enum` is pretty much syntactic sugar for sealed families (not much reason to use sealed families instead of enums in Scala 3 unless you have some very special case where enum doesn't work, so enum and sealed family should be treated as the same thing), you'd have to first try do this derivation to see if all types of the family are objects (no case classes in the enum) and then try to do the other derivations if that fails (i.e. within the `derived` method the "normal" derivation would act as a fallback if the enum derivation fails). It should be possible to do that, but then it would also not be possible to choose the "normal" derivation for sums if the sum consists of only objects and no case classes (but this might not be a big loss).

Which approach is preferable here? Use `ConfigReader` everywhere and make the choice of derivation for enums "automatic", or keep the choice and also keep the `EnumConfigReader` type?

Reply:
If `enum` is just syntactic sugar for sealed families, then I totally agree with not making things automatic - it would be a departure from existing behavior and it could lead to inconsistencies where adding a new option would change completely the expected format.